src/polyswarm_api/api.py

```diff
@@ -669,9 +669,38 @@ def download_id(self, out_dir, instance_id):
 
         return artifact
 
-    def sandbox(self, instance_id):
-        logger.info('Sandboxing %s', instance_id)
-        return resources.SandboxTask.create(self, artifact_id=instance_id).result()
+    def sandbox(self, instance_id, sandbox):
+        logger.info('Sandboxing %s in %s', instance_id, sandbox)
+        return resources.SandboxTask.create(self, artifact_id=instance_id, sandbox=sandbox).result()
+
+    def sandbox_file(self, artifact, sandbox, artifact_type=resources.ArtifactType.FILE, artifact_name=None):
+        logger.info('Sandboxing file in %s', sandbox)
+        artifact_type = resources.ArtifactType.parse(artifact_type)
+        # TODO This is a python 2.7 check if artifact is a file-like instance, consider changing
+        #  to isinstance(artifact, io.IOBase) when deprecating 2.7 and implementing making LocalHandle
+        #  inherit io.IOBase, although this will change the method delegation logic in the resource
+        if hasattr(artifact, 'read') and hasattr(artifact.read, '__call__'):
```

Comment:
Is better use the `callable(func)` check, instead of explicitly looking for `__call__` existence, as `callable()` handle more cases, including test mocks and monkeypatches 
```suggestion
        if hasattr(artifact, 'read') and callable(artifact.read):
```

Reply:
This is a python 3 method, I don't want to handle any python 2 deprecation issues right now. Once we decide to remove old python 2 related stuff we can do this.